keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1116 @@
+CoreDNS for example). Those traffic should not be blocked when developers apply
+NetworkPolicy to their Namespaces which isolates the workloads.
+
+AdminNetworkPolicy `Pass` rules allows an admin to delegate security posture for
+certain traffic to the Namespace owners by overriding any lower priority Allow or 
+Deny rules. For example, allowing intra-tenant traffic can be delegated to tenant 
+admins explicitly by the cluster admin with the use of `Pass` rules.
+
+#### Priority
+
+The policy instances will be ordered based on the numeric priority assigned to the
+ANP. A larger number corresponds to a higher precedence. For example, ANP with `priority`
+set to "1" will be considered as the policy with the lowest precedence, while "999" will
+be considered as highest precedence. Any positive priority, however, will have
+higher precedence over the namespaced NetworkPolicy instances in the cluster.
+
+Additionally, the special priority "0" can be used in the priority field to indicate
+that the rules in that policy instance shall be created at a priority lower than the
+Namespaced NetworkPolicies. 
+
+The relative precedence of the rules within a ANP will be determined by the order in which the
+rule is written. Thus, a rule that appears at the top of the ingress/egress rules would take the
+highest precedence.
+
+Conflict resolution: Two policies are considered to be conflicting if they are assigned the
+same `priority` and apply to the same resources or a union of resources. In order to avoid such conflicts, 
+we propose to include tooling for ANP resources to help alert the admin to potentially ambiguous ANP
+priority scenarios, more details in [risks and mitigations](#risks-and-mitigations). However, ultimately
+it will be the job of the CNI to decide how to handle overlapping priority situations in the respective 
+implementations. 
+
+For conformance and portability purposes, it will be required that implementations
+support the range of 1000 numeric ANP priorities (0 through 999) per cluster and a maximum of 100 rules
+within an ANP instance. This ensures that policy manifests written using priorities
+0 through 999 are portable across clusters and implementations without the
+need to re-write policy manifests or re-adjust priority allocations and also ensures we maintain 
+safe upper bounds. 
+
+#### Multiple AppliedTos per policy instance
+
+With the current Kubernetes NetworkPolicy v1 api, an instance of a policy can be applied
+to a single set of pods defined using a single selector instance.
+However with the proposed ANP specification it shall be possible to apply the policy to 
+multiple unique sets of Namespaces/pods within a single ANP policy instance, where each such set can 
+be configured with its own set of peer endpoints and policy actions. This enables greater flexibility
+of configuration, and improves ease of use by not requiring cluster admins to have to come up
```

Comment:
(Without having thought about this too deeply,) I feel like it would be nice if each ANP represented a single policy "goal". eg, one for "tenants can talk to themselves, but not each other by default", and one for "everyone can always talk to CoreDNS", and one for "the monitoring namespace can always talk to everyone" and one for "nobody is allowed to talk to Service X, except for Alice and Bob".

(So the last one might need multiple targets, but none of the others would.)

Reply:
> I feel like it would be nice if each ANP represented a single policy "goal"

I agree. I don't know how to structure the API to force users to do that, but I feel like allowing "ApplyTo" per-rule removes one of the guardrails which nudges them that direction.

Reply:
This convo is also happening above https://github.com/kubernetes/enhancements/pull/2522#discussion_r791174776

Reply:
.Does the last one need multiple targets?  Only if you want to specify both the ingress-to-X and egress-from-everyone rules

Reply:
ah, you're right